.github/workflows/unit-tests.yaml

```diff
@@ -0,0 +1,200 @@
+        username: ${{ secrets.DOCKERHUB_USER }}
+        password: ${{ secrets.DOCKERHUB_TOKEN }}
+    env:
+      RUSTFLAGS: ${{ matrix.rustflags }}
+      VYPER_DOWNLOAD_EXTENSION: ".linux"
+    steps:
+      - name: Checkout source
+        uses: actions/checkout@v4
+      - name: Prepare environment
+        run: |
+          rustup target add ${{ matrix.target }}
+      - name: Install vyper compiler
+        run: |
+          curl --verbose \
+            --location -o "${PWD}/vyper" \
+            "${VYPER_DOWNLOAD_URL}/${VYPER_VERSION}/${VYPER_DOWNLOAD_FILENAME}${VYPER_DOWNLOAD_EXTENSION}" && \
+            chmod a+x "${PWD}/vyper"
+          echo "${PWD}" >> "${GITHUB_PATH}"
+      - name: Clone LLVM framework
+        run: |
+          cargo install compiler-llvm-builder --target ${{ matrix.target }}
+          zkevm-llvm clone
+      - name: Build LLVM framework
+        uses: nick-fields/retry@v2
+        with:
+          timeout_minutes: 60
+          max_attempts: 2 # protection mechanism for MUSL sporadic download failure
+          command: zkevm-llvm build
+      - name: Build zkvyper
+        run: |
+          cargo build -vv --release --target ${{ matrix.target }}
+          echo "${PWD}/target/${{ matrix.target }}/release" >> "${GITHUB_PATH}"
+      - name: Run unit tests
+        run: |
+          cargo install cargo2junit
+          cargo test --verbose --target ${{ matrix.target }} -- -Z unstable-options --format json | cargo2junit > "${UNIT_TESTS_RESULTS_XML}"
+          cat "${UNIT_TESTS_RESULTS_XML}"
+      - name: Upload test results
+        if: always()
+        uses: EnricoMi/publish-unit-test-result-action@v2
+        with:
+          check_name: ${{ matrix.name }} Unit Tests Results
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          files: ${{ env.UNIT_TESTS_RESULTS_XML }}
+          action_fail_on_inconclusive: true
+
+  unit-tests-windows:
+    name: "Windows"
+    runs-on: windows-2022-github-hosted-16core
+    env:
+      LIBSTDCPP_SOURCE_PATH: "C:/a/_temp/msys64/mingw64/lib/libstdc++.a"
+      VYPER_DOWNLOAD_EXTENSION: ".windows.exe"
+    defaults:
+      run:
+        shell: msys2 {0}
+    steps:
+      - name: Checkout source
+        uses: actions/checkout@v4
+      - name: Prepare msys2
+        uses: msys2/setup-msys2@v2
+      - name: Prepare env
+        run: |
+          pacman-key --refresh
+          pacman -Sy
+          curl -LO https://repo.msys2.org/mingw/mingw64/mingw-w64-x86_64-cmake-3.27.7-3-any.pkg.tar.zst
+          pacman --noconfirm -U mingw-w64-x86_64-cmake-3.27.7-3-any.pkg.tar.zst
+          pacman --noconfirm -S --needed --overwrite \
+          base-devel \
+          git \
+          ninja \
+          mingw-w64-x86_64-clang \
+          mingw-w64-x86_64-lld \
+          mingw-w64-x86_64-rust \
+          mingw-w64-x86_64-gcc-libs \
+          mingw-w64-x86_64-gcc
+      - name: Install vyper compiler
+        run: |
+          curl --verbose --location -o vyper.exe \
```

Comment:
I’d also prefer not to see `--verbose` output I normally don’t care about.

Reply:
I usually prefer to add `--verbose` to `curl` to get more information if something is wrong with the download, otherwise, it can be not easy to debug later. But for now, I believe it's ok to remove. I did this in my update.

Reply:
It’s a tradeoff, of course.  But now that you’ve got a button for a manual run, it’s easy enough to add  `--verbose` in a branch when needed.